src/main/java/edu/unc/lib/boxc/migration/cdm/services/PermissionsService.java

```diff
@@ -0,0 +1,159 @@
+import edu.unc.lib.boxc.migration.cdm.exceptions.StateAlreadyExistsException;
+import edu.unc.lib.boxc.migration.cdm.model.DestinationsInfo;
+import edu.unc.lib.boxc.migration.cdm.model.MigrationProject;
+import edu.unc.lib.boxc.migration.cdm.model.PermissionsInfo;
+import edu.unc.lib.boxc.migration.cdm.options.PermissionMappingOptions;
+import edu.unc.lib.boxc.migration.cdm.util.ProjectPropertiesSerialization;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.csv.CSVRecord;
+import org.slf4j.Logger;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Service for permission mapping
+ * @author krwong
+ */
+public class PermissionsService {
+    private static final Logger log = getLogger(PermissionsService.class);
+
+    private MigrationProject project;
+    private String everyoneField;
+    private String authenticatedField;
+    private List<String> patronRoles;
+
+    /**
+     * Generates default mapping csv
+     * @param options default mapping options
+     */
+    public void generateDefaultPermissions(PermissionMappingOptions options) throws Exception {
+        Path fieldsPath = project.getPermissionsPath();
+        ensureMappingState(options.isForce());
+
+        List<String> patronRoles = getPatronRoles();
+
+        // Everyone permission
+        if (options.isStaffOnly()) {
+            everyoneField = "none";
+        } else if (options.getEveryone() != null) {
+            if (patronRoles.contains(options.getEveryone())) {
+                everyoneField = options.getEveryone();
+            } else {
+                throw new IllegalArgumentException("Everyone value is invalid. " +
+                        "Must be one of the following patron roles: " + patronRoles);
+            }
+        } else {
+            // if no permissions/roles are specified, default to canViewOriginals
+            everyoneField = "canViewOriginals";
+        }
+
+        // Authenticated permission
+        if (options.isStaffOnly()) {
+            authenticatedField = "none";
+        } else if (options.getAuthenticated() != null) {
+            if (patronRoles.contains(options.getAuthenticated())) {
+                authenticatedField = options.getAuthenticated();
+            } else {
+                throw new IllegalArgumentException("Authenticated value is invalid. " +
+                        "Must be one of the following patron roles: " + patronRoles);
+            }
+        } else {
+            // if no permissions/roles are specified, default to canViewOriginals
+            authenticatedField = "canViewOriginals";
+        }
```

Comment:
To dry this up some, this block is the same as the everyone block except with Authenticated instead of Everyone, so you could make a helper method that takes isStaffOnly and the value of the user role option, and returns the role to use. 
```
authenticatedField = getAssignedRoleValue(options.isStaffOnly(), options.getAuthenticated())
```
Or the helper could just take the value of the user role option, and then all of the isStaffOnly assignments could happen together.
```
if (options.isStaffOnly()) {
     everyoneField = "none";
     authenticatedField = "none";
} else {
    everyoneField = getAssignedRoleValue(options.getEveryone());
    authenticatedField = getAssignedRoleValue(options.getAuthenticated());
}
```